llpc/util/llpcGfxRegHandler.h

```diff
@@ -0,0 +1,128 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2019-2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+ /**
+  ***********************************************************************************************************************
+  * @file  llpcGfxRegHandler.h
+  * @brief LLPC header file: contains the definition of LLPC utility class GfxIpRegHandlerBase.
+  ***********************************************************************************************************************
+  */
+
+#pragma once
+#include "llpcBuilderImpl.h"
+#include "llpcPipelineState.h"
+
+namespace Llpc
+{
+
+using namespace llvm;
+
+// General bits info for indexed DWORD
+struct BitsInfo
+{
+    uint32_t index;
+    uint32_t offset;
+    uint32_t count;
+};
+
+// =====================================================================================================================
+// Base class for handling GFX-specific registers
+class GfxIpRegHandlerBase
```

Comment:
I forget to point out this: GfxIpRegHandlerBase -> GfxRegHandlerBase

Reply:
Okay, I'll  update it soon.